docs/lab/00.md

```diff
@@ -0,0 +1,517 @@
+
+```rust
+fn main() {
+
+    let name = "Mary";
+    let age = 26;
+
+    println!("Hello, {}. You are {} years old", name, age);
+    // if the replacements are only variable, one can use the inline version
+    println!("Hello, {name}. You are {age} years old");
+}
+```
+
+### Variables and mutability
+
+We use the `let` keyword to create a variable.
+
+```rust
+    let a = 5;
+```
+
+By default, in Rust, variables are **immutable** , meaning once a value is tied to a name, you cannot
+change that value.
+
+Example:
+
+```rust
+fn main() {
+    let x = 5;
+    println!("The value of x is: {x}");
+    x = 6;
+    println!("The value of x is: {x}");
+}
+```
+
+In this case, we will get a compilation error because we are trying to modify the value of `x` from `5` to `6`, but
+`x` is immutable, so we cannot make this modification.
+
+Although variables are immutable by default, you can make them **mutable** by adding `mut` in front of the
+variable name. Adding `mut` also conveys intent to future readers of the code by indicating that other parts
+of the code will modify the value of this variable.
+
+```rust
+fn main() {
+    let mut x = 5;
+    println!("The value of x is: {x}");
+    x = 6;
+    println!("The value of x is: {x}");
+}
+```
+
+Now the value of `x` can become `6`.
+
+### Constants
+
+Like immutable variables, constants are values that are tied to a name and **are not allowed to change**,
+but there are some differences between constants and variables.
+
+First of all, you are not allowed to use `mut` with constants. Constants are not only immutable by default,
+they are always immutable. You declare constants using the `const` keyword instead of the let keyword .
+
+```rust
+const THREE_HOURS_IN_SECONDS: u32 = 60 * 60 * 3;
+```
+
+:::info 
+
+For a better understanding, please read [chapter 3](https://doc.rust-lang.org/book/ch03-00-common-programming-concepts.html) of the documentation.
+
+:::
+
+### Data Types
+
+#### Scalar types
+
+A scalar type represents a single value. Rust has four main scalar types: integers, floating point numbers,
+booleans, and characters.
+
+**Integer** → Each variant can be signed or unsigned and has an explicit size.
+```rust
+let x: i8 = -2;
+let y: u16 = 25;
+```
+
+| Length              | Signed | Unsigned | Java Equivalent |
+| :----------------: | :------: | :----: | :----: |
+| 8-bit |   `i8`   | `u8` | `byte`/ `Byte`[^java_unsigned] |
+| 16-bit |   `i16`   | `u16` | `short` / `Short`[^java_unsigned] |
+| 32-bit |   `i32`   | `u32` | `int` / `Integer`[^java_unsigned] |
+| 64-bit |   `i64`   | `u64` | `long` / `Long`[^java_unsigned] |
+| 128-bit |   `i128`   | `u128` | N/A |
+| arch |   `isize`   | `usize` | N/A |
+
+**Floating Point** → Rust's floating point types are f32 and f64, which are 32-bit and 64-bit in size, respectively. The default type is **f64** because on modern CPUs it is about the same speed as f32 but is capable of more precision. All floating point types are **signed**.
```

Comment:
```suggestion
**Floating Point** → Rust's floating point types are `f32` and `f64`, which are 32-bit and 64-bit in size, respectively. The default type is `f64` because on modern CPUs it is about the same speed as `f32` but is capable of more precision. All floating point types are **signed**.
```